catalog/ui/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/suggestion/MgrsCoordinateProcessorTest.java

```diff
@@ -0,0 +1,360 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.query.suggestion;
+
+import static org.codice.ddf.catalog.ui.query.suggestion.LatLon.from;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.both;
+import static org.hamcrest.Matchers.contains;
+import static org.hamcrest.Matchers.hasSize;
+
+import java.util.Arrays;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.codice.ddf.spatial.geocoding.Suggestion;
+import org.hamcrest.Description;
+import org.hamcrest.TypeSafeMatcher;
+import org.junit.Test;
+
+public class MgrsCoordinateProcessorTest {
```

Comment:
✏️ Unit tests for cases where exceptions are thrown

Reply:
Not actually possible. There's one `IllegalArgumentException` I'm using, but no input can actually trigger it. `PATTERN_MGRS_ANYPART` is an OR'd sequence of each regex mapping to an MGRS part type. By definition if it found a match, it will be either `ZONE`, `HUNDRED_KM`, or `NUMERIC`. The `NONE` value is there to explicitly handle the initial state. The trade-off is we need to handle it in the switch but it's an impossible case so we throw an exception.